Better Biogenerator

Missile597 opened the pull request:
## About The Pull Request

Adds some more items to the biogenerator to bridge the gap in botany with some things that are needed/highly desirable in that department. Aim was to make the biogenerator (something that barely gets used at the moment) more useful without just eclipsing chemistry or cargo completely.

## Changelog
:cl:
add: Added new materials to the biogenerator list: Diethylamine, Unstable Mutagen, Plant-B-Gone, Table Salt, Soy Sauce, Flora Disk, Floral Somatoray, Kudzu Seeds, Random Exotic Seeds
balance: Lowered the costs of the fertilisers and botany materials to make them easier to obtain. After all, the department uses these a lot.
/:cl:

Missile597 commented:
This is also an indirect solution to solve the suggestion raised in #9625 outside of mapping, giving botanists a more interesting source of fertiliser which doesn't just totally eclipse chemistry or the existing biogenerator chems

Kashargul reviewed (changes requested):
Surround additions and edits with

// CHOMPEdit Start
// CHOMPEdit End

//CHOMPAdd Start
//CHOMPAdd End

Kashargul commented on code/game/machinery/biogenerator.dm:
```diff
@@ -76,15 +76,43 @@
 		BIOGEN_ITEM("Nutri-spread", /obj/item/reagent_containers/food/snacks/spreads, 1, 30),
 		BIOGEN_ITEM("Nutri-spread x5", /obj/item/reagent_containers/food/snacks/spreads, 5, 120),
+		//CHOMPAdd Start - adding common condiments to the list for chefs
+		BIOGEN_REAGENT("Table Salt x10", REAGENT_ID_SODIUMCHLORIDE, 10, 20),
```
REAGENT_SODIUMCHLORIDE + " x10"

Kashargul commented on code/game/machinery/biogenerator.dm:
```diff
@@ -76,15 +76,43 @@
 		BIOGEN_ITEM("Nutri-spread x5", /obj/item/reagent_containers/food/snacks/spreads, 5, 120),
+		//CHOMPAdd Start - adding common condiments to the list for chefs
+		BIOGEN_REAGENT("Table Salt x10", REAGENT_ID_SODIUMCHLORIDE, 10, 20),
+		BIOGEN_REAGENT("Table Salt x50", REAGENT_ID_SODIUMCHLORIDE, 50, 100),
```
^

Kashargul commented on code/game/machinery/biogenerator.dm:
```diff
@@ -76,15 +76,43 @@
+		//CHOMPAdd Start - adding common condiments to the list for chefs
+		BIOGEN_REAGENT("Table Salt x10", REAGENT_ID_SODIUMCHLORIDE, 10, 20),
+		BIOGEN_REAGENT("Table Salt x50", REAGENT_ID_SODIUMCHLORIDE, 50, 100),
+		BIOGEN_REAGENT("Soy Sauce x10", REAGENT_ID_SOYSAUCE, 10, 30),
```
REAGENT_SOYSAUCE + " x10"

Kashargul commented on code/game/machinery/biogenerator.dm:
```diff
@@ -76,15 +76,43 @@
+		BIOGEN_REAGENT("Table Salt x10", REAGENT_ID_SODIUMCHLORIDE, 10, 20),
+		BIOGEN_REAGENT("Table Salt x50", REAGENT_ID_SODIUMCHLORIDE, 50, 100),
+		BIOGEN_REAGENT("Soy Sauce x10", REAGENT_ID_SOYSAUCE, 10, 30),
+		BIOGEN_REAGENT("Soy Sauce x50", REAGENT_ID_SOYSAUCE, 50, 120),
```
^

Kashargul commented on code/game/machinery/biogenerator.dm:
```diff
@@ -76,15 +76,43 @@
+		BIOGEN_ITEM("Robust Harvest x5", /obj/item/reagent_containers/glass/bottle/robustharvest, 5, 250),
+		//CHOMPedit End
+		//CHOMPAdd Start - adding additional fertilisers for botany use
+		BIOGEN_ITEM("Diethylamine", /obj/item/reagent_containers/glass/bottle/diethylamine, 1, 60),
```
As above. Newly added content should use defines.

Missile597 replied:
> As above. Newly added content should use defines.

These options don't spawn reagents, they spawn reagent containers. They are BIOGEN_ITEM not BIOGEN_REAGENT

All the BIOGEN_REAGENT should be using the new defines

Missile597 commented:
Should be good now. Had to redo your CHOMPEdit capitalisation change as I forgot to pull first but should all be the same if you want to double check. Tested the defines didn't break the UI and it worked fine

Cyrelius commented:
Please remove the entries about making disks and Somatorays, these should be created in an autolathe or protolathe, not a bioprinter.

	item_list["Botany Tools"] = list(
		BIOGEN_ITEM("Flora Disk", /obj/item/disk/botany, 1, 100),
		BIOGEN_ITEM("Floral Somatoray", /obj/item/gun/energy/floragun, 1, 1000),

Missile597 commented:
> Please remove the entries about making disks and Somatorays, these should be created in an autolathe or protolathe, not a bioprinter.
> 
> ```
> item_list["Botany Tools"] = list(
> 	BIOGEN_ITEM("Flora Disk", /obj/item/disk/botany, 1, 100),
> 	BIOGEN_ITEM("Floral Somatoray", /obj/item/gun/energy/floragun, 1, 1000),
> ```

Done. Pretty sure the flora disks can't be made atm which is why I added them, but for now I just removed them from the list. Can always look at adding them to the protolathe list some other time

Kashargul commented:
> > Please remove the entries about making disks and Somatorays, these should be created in an autolathe or protolathe, not a bioprinter.
> > ```
> > item_list["Botany Tools"] = list(
> > 	BIOGEN_ITEM("Flora Disk", /obj/item/disk/botany, 1, 100),
> > 	BIOGEN_ITEM("Floral Somatoray", /obj/item/gun/energy/floragun, 1, 1000),
> > ```
> 
> Done. Pretty sure the flora disks can't be made atm which is why I added them, but for now I just removed them from the list. Can always look at adding them to the protolathe list some other time

The gun is from the protolathe, the disks can be ordered by cargo
